packages/smooth_app/lib/pages/product/product_image_gallery_view.dart

```diff
@@ -259,15 +265,11 @@ class _ProductImageGalleryViewState extends State<ProductImageGalleryView> {
       images
           .groupListsBy((ProductImage element) => element.imgid)
           .values
-          .map((List<ProductImage> sameIdImages) => sameIdImages.firstOrNull)
+          .map(_selectProductImage)
           .whereNotNull();
 
-  /// Created a [ProductImageData] from a [ProductImage]
-  ProductImageData _getProductImageData(ProductImage image) => ProductImageData(
-        imageField: image.field,
-        // TODO(VaiTon): i18n
-        title: image.imgid ?? '',
-        buttonText: image.imgid ?? '',
-        imageUrl: ImageHelper.buildUrl(widget.product.barcode, image),
-      );
+  ProductImage? _selectProductImage(Iterable<ProductImage> images) =>
+      images.firstWhereOrNull(
+          (ProductImage image) => image.size == ImageSize.DISPLAY) ??
```

Comment:
@WildOrangutan So if I sum up, that's where you deduplicate by restricting to `400` pictures only, right?

I would rename `_selectProductImage` to `_selectBestProductImage`, and/or add a comment.
I believe the point of this method should be to take the _best_ picture: `400`, or `200`, or `100`, and that's not exactly what is in your code (you just say `400` or else whatever).
Perhaps you could rewrite it in more than one line to make it explicit (we have a method here, we have a bit of space): that would help the code reviewer now (it's me) and whoever will maintain the code later.
You can even write it in one line:
```dart
images.firstWhereOrNull(
           (ProductImage image) => image.size == ImageSize.DISPLAY) ??
images.firstWhereOrNull(
           (ProductImage image) => image.size == ImageSize.SMALL) ??
images.firstWhereOrNull(
           (ProductImage image) => image.size == ImageSize.THUMB)
```

Reply:
> So if I sum up, that's where you deduplicate by restricting to 400 pictures only, right?
Yes, this is where I selected 400 as middle ground between two screens (1st and 2nd screenshot).

I've updated the code, hope I didn't complicate things to much, by using map.

Reply:
Also, what about `ORIGINAL`, `UNKOWN` or `null` image sizes? Isn't it better to image load too slow, than not load at all?

Reply:
@WildOrangutan If for some reason we have to download hundreds of "other" `ORIGINAL`, that can be ugly, performance-wise. Could even crash the app I guess.

But
* that's theoretical, as we can expect `400` versions everywhere
* in your latest code no additional comment is needed: anybody that reads it can understand the priorities and - if relevant later - clearly remove one picture version or reorder them (e.g. "`200` is good enough" or "never `full` or `unknown`").

So let's go with it!